### PR TITLE
Added tests show and hide items for score board page

### PR DESCRIPTION
**Pull request by InnaO**

### CODE REVIEW

**Comment by InnaO** [comment]
@IgorSasovets Please, look at my PR?

File: pages/basePage.js [hunk 1]
```diff
@@ -1,37 +1,77 @@
...
-    const processedElement = typeof element === 'object' ? await element : element;
-    let iteration = 0;
-    do {
-        try {
-            await browser.waitUntil(() => processedElement.isDisplayed(), timeout);
-            return;
-        } catch (err) {
-            iteration++;
-            console.log(err);
+    async waitForElement(element, { timeout = 5000, retries = 10 } = {}) {
```
Comment (IgorSasovets) [inline pages/basePage.js]: Please remove unnecessary white spaces

File: tests/score-board.spec.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+const assert = require('assert');
+const ScoreBoardPage = require('../pages/JuiceShop/ScoreBoard');
+
+describe('Hide and show all score board challenge items', () => {
+    before(async () => {
```
Comment (IgorSasovets) [inline tests/score-board.spec.js]: Please remove white space after `async`, the same is actual for the `it` blocks descriptions

**Review by IgorSasovets — CHANGES_REQUESTED** [review]
@InnaO , good job. Please add minor fixes and it will be resolved as done

**Review by InnaO — COMMENTED** [review]
@IgorSasovets I have added fixes to my PR. Please, review

File: pages/basePage.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
 /* eslint-disable class-methods-use-this */
 'use strict';
-
```
Comment (IgorSasovets) [inline pages/basePage.js:2]: Why did you remove this line?

Comment (IgorSasovets) [reply pages/basePage.js:2]: Also, change
```
async waitForElement(element, { timeout = 5000, retries = 10 } = {})
```
to
```
async waitForElement(element, {timeout = 5000, retries = 10} = {})
```

File: tests/score-board.spec.js [hunk 4]
```diff
@@ -3,19 +3,19 @@ const assert = require('assert');
 const ScoreBoardPage = require('../pages/JuiceShop/ScoreBoard');
 
 describe('Hide and show all score board challenge items', () => {
-    before(async () => {
+    before(async() => {
         await ScoreBoardPage.navigateToScoreBoardPage();
         await ScoreBoardPage.closeWelcomeBanner();
     });
-    it('Should change challenge items visibility to hide all ', async () => {
```
Comment (IgorSasovets) [inline tests/score-board.spec.js]:
```
it('Should change challenge items visibility to hide all ', async() => {
```
to
```
it('Should change challenge items visibility to hide all', async() => {
```

**Review by IgorSasovets — CHANGES_REQUESTED** [review]
@InnaO , please see my comments

**Review by InnaO — COMMENTED** [review]
@IgorSasovets sorry, it was my mistake
Please, check again

**Review by IgorSasovets — APPROVED** [review]

**Pull request merged**